### PR TITLE
Chore: Avoid explicit React.FC<Props> when possible (round 2)

### PR DESCRIPTION
**Pull request by ryantxu**
Continuing... https://github.com/grafana/grafana/pull/64722

This PR replaces most explicit uses of React.FC with a typed function.

The explicit FC components exist everywhere so are often used as the copy from source :(

See:
* https://medium.com/raccoons-group/why-you-probably-shouldnt-use-react-fc-to-type-your-react-components-37ca1243dd13
* https://github.com/facebook/create-react-app/pull/8177

### CODE REVIEW

**Review by aangelisc — APPROVED** [review]
Azure changes LGTM 😊

**Review by nmarrs — APPROVED** [review]
⭐️

File: packages/grafana-ui/src/components/DateTimePickers/TimeZonePicker/TimeZoneOption.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ export interface SelectableZone extends SelectableValue<string> {
   searchIndex: string;
 }
 
-export const WideTimeZoneOption: React.FC<PropsWithChildren<Props>> = (props, ref) => {
+export const WideTimeZoneOption = (props: PropsWithChildren<Props>) => {
```
Comment (jackw) [inline packages/grafana-ui/src/components/DateTimePickers/TimeZonePicker/TimeZoneOption.tsx:29]: I've not come across `PropsWithChildren` before. Is it something we want to keep? (Not suggesting removing it in this PR, more asking for my own understanding.)

**Review by jackw — APPROVED** [review]
LGTM! 🚀

File: packages/grafana-ui/src/components/DateTimePickers/TimeZonePicker/TimeZoneOption.tsx (hunk 1 above)
Comment (ryantxu) [reply packages/grafana-ui/src/components/DateTimePickers/TimeZonePicker/TimeZoneOption.tsx:29]: We use it in a few places... it is just a buildin react interface that has children specified

**Pull request merged**